types/parameter_types/abstract_parameter_type.go

```diff
@@ -0,0 +1,23 @@
+package parameter_types
```

Comment:
since this is already in the `types` package, could we shorten this package name to just `parameters` ?

Also need to add the license identifier header

Reply:
added license
renamed this package to "leaf_parameters" since it now only contains leaf parameters